src/protected/application/lib/MapasCulturais/Traits/ControllerAPI.php

```diff
@@ -123,9 +123,24 @@ public function API_find(){
         $this->apiResponse($data);
     }
 
-    public function API_describe(){
+    public function API_describe(){        
         $class = $this->entityClassName;
-        $this->apiResponse($class::getPropertiesMetadata());
+        $data_array = $class::getPropertiesMetadata();                
+        $file_groups = App::i()->getRegisteredFileGroupsByEntity( $class );
```

Comment:
Você tem certeza que aqui é sempre retornado um array? será q não tem algum caso que não tenha @files e isso retorne vazio? acho que vale uma checagem ali antes do foreach, pro caso de não vir um array

Reply:
O método `getRegisteredFileGroupsByEntity($entity)` realiza uma comparação no seu retorno:

`return $controller_id && key_exists($controller_id, $this->_register['file_groups']) ? $this->_register['file_groups'][$controller_id] : [];`

Dessa forma se não existir um **@files** para a entidade é retornado um array vazio. Nesse casso o resultado do Json vai ser:
```
... },
    "@file": []
}
```

Reply:
ok!